raiden/channel/participant_state.py

```diff
@@ -3,44 +3,40 @@
 
 from raiden.mtree import Merkletree
 from raiden.utils import sha3
-from raiden.channel.balance_proof import BalanceProof
 
 log = slogging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
 class ChannelEndState(object):
     """ Tracks the state of one of the participants in a channel. """
 
-    def __init__(
-            self,
-            participant_address,
-            participant_balance,
-            opened_block,
-            transferred_amount=0):
+    def __init__(self,
+                 participant_address,
+                 participant_balance,
+                 balance_proof):
+
         # since ethereum only uses integral values we cannot use float/Decimal
         if not isinstance(participant_balance, (int, long)):
             raise ValueError('participant_balance must be an integer.')
 
-        if not opened_block > 0:
-            raise ValueError('opened_block must be non-zero and positive (not %s)' % opened_block)
-
         self.contract_balance = participant_balance
         self.address = participant_address
 
-        # amount of token transferred and unlocked
-        self.transferred_amount = transferred_amount
-
-        # Sequential nonce, current value has not been used, 0 must not be used
-        # since in the netting contract it represents null.
-        #
-        # The nonce value must be inside the netting channel allowed range
-        # that is defined in terms of the opened block
-        self.nonce = 1 * (opened_block * (2 ** 32))
-
         # contains the last known message with a valid signature and
         # transferred_amount, the secrets revealed since that transfer, and the
         # pending locks
-        self.balance_proof = BalanceProof()
+        self.balance_proof = balance_proof
+
+    def transferred_amount(self, other):
```

Comment:
I find this interface a little counterintuitive (especially since it was a plain attribute before).

1: how about naming it `transferred_amount_of(other_state)` ? That way it is easily distinguished from the (still attribute) `balance_proof.transferred_amount`.

2: why is this a method of the "from"-state? It does not need `self` and could as well be used in a weird way like
```
self.transferred_amount(self, self)
```
I have the feeling, it should still be a property and would work the other way around:
```
ChannelEndState.get_transferred_amount(self):
  if self.balance_proof.balance_proof:
    return self.balance_proof.balance_proof.transferred_amount
  return 0
```

Usage example from above (https://github.com/raiden-network/raiden/pull/913/files#diff-263eac5cb6d265279d2898e4c2bb953fR543) would be:

```
transferred_amount = to_state.get_transferred_amount() + lock.amount
```

Reply:
This is fixed by the follow up PR: https://github.com/raiden-network/raiden/pull/932